### PR TITLE
Add function display_and_save_batch in wenetspeech/pruned_transducer_stateless2/train.py

### PR DESCRIPTION
**Pull request by yangsuxia**
Add function display_and_save_batch in egs/wenetspeech/ASR/pruned_transducer_stateless2/train.py

### CODE REVIEW

File: egs/wenetspeech/ASR/pruned_transducer_stateless2/train.py [hunk 1]
```diff
@@ -957,6 +961,34 @@ def remove_short_and_long_utt(c: Cut):
...
+    """Display the batch statistics and save the batch into disk.
+
+    Args:
+      batch:
+        A batch of data. See `lhotse.dataset.K2SpeechRecognitionDataset()`
+        for the content in it.
+      params:
+        Parameters for training. See :func:`get_params`.
+      sp:
+        The BPE model.
```
Comment (csukuangfj) [inline egs/wenetspeech/ASR/pruned_transducer_stateless2/train.py:977]:
```suggestion
```

File: egs/wenetspeech/ASR/pruned_transducer_stateless2/train.py [hunk 2]
```diff
@@ -957,6 +961,34 @@ def remove_short_and_long_utt(c: Cut):
...
+    filename = f"{params.exp_dir}/batch-{uuid4()}.pt"
+    logging.info(f"Saving batch to {filename}")
+    torch.save(batch, filename)
+
+    supervisions = batch["supervisions"]
+    features = batch["inputs"]
+
+    logging.info(f"features shape: {features.shape}")
+
+    num_tokens = params.vocab_size
```
Comment (csukuangfj) [inline egs/wenetspeech/ASR/pruned_transducer_stateless2/train.py:990]: https://github.com/k2-fsa/icefall/blob/a2a79c7992f8ef45e1843d419236c5b0b3eb5ac8/egs/wenetspeech/ASR/pruned_transducer_stateless2/train.py#L573-L575

Please print the number of elements in `y`.

Comment (yangsuxia) [reply egs/wenetspeech/ASR/pruned_transducer_stateless2/train.py]: > https://github.com/k2-fsa/icefall/blob/a2a79c7992f8ef45e1843d419236c5b0b3eb5ac8/egs/wenetspeech/ASR/pruned_transducer_stateless2/train.py#L573-L575
> 
> Please print the number of elements in `y`.

I see, I have modified num_tokens in display_and_save_batch

File: egs/wenetspeech/ASR/pruned_transducer_stateless2/train.py (hunk 1 above)
Comment (yangsuxia) [reply egs/wenetspeech/ASR/pruned_transducer_stateless2/train.py]: I see, I have modified num_tokens in display_and_save_batch

**Comment by csukuangfj** [comment]
Thanks!

**Comment by csukuangfj** [comment]
Could you fix the style issues reported by GitHub actions?

You can refer to https://k2-fsa.github.io/icefall/contributing/code-style.html
to check your code style locally.

**Comment by csukuangfj** [comment]
Thanks!

**Pull request merged**